### PR TITLE
General report

### PR DESCRIPTION
**Pull request by g1st**
### Proposal
A work in progress.

### Checklist

- [x] I have made this pull request to the `master` branch
- [x] I have run `npm run lint` and there are no errors

### CODE REVIEW

**Comment by g1st** [comment]
I think everything works ok for general reports and it looks like the [suggested format](https://docs.google.com/spreadsheets/d/1WSKei3BWWrTwyypKUOjE1uAdKRjOgIjThs0BPAlbQOA/edit#gid=0).
I am not sure if in the navigation dropdown General report should go first or last. Also, the table isn't looking very nice when there are comments on it but that's how the client wants it :shrug:

**Comment by GTabala** [comment]
I think we can add a "Button" like "Reset" before that will hide the feedback column and show it back (Using names like "Hide feedback" and "Show feedback").

File: client/src/components/Reports/GeneralTable.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		const m = Math.abs(minutes).toString().padStart(2, "0");
+
+		return `${hours < 0 || minutes < 0 ? "-" : " "}${h}:${m}`;
+	};
+
+	const handleClick = (id) => {
+		history.push(`/edit-jobs/${id}`);
+	};
+
+	const handleKeyPress = (id, customer, e) => {
```
Comment (GTabala) [inline client/src/components/Reports/GeneralTable.js:20]: "customer" is redundant.

File: client/src/components/Reports/GeneralTable.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+							{formatDuration(contracted_duration)}
+						</td>
+						<td className={tableFooter && "font-weight-bold"}>
+							{formatDuration(actual_duration)}
+						</td>
+						<td className={tableFooter && "font-weight-bold"}>
+							{formatDuration(difference)}
+						</td>
+						<td className={tableFooter && "font-weight-bold"}>{feedback}</td>
+					</tr>
```
Comment (GTabala) [inline client/src/components/Reports/GeneralTable.js:79]: There is no sense to use `"font-weight-bold"` when data is `undefined`. Check other lines.

**Comment by g1st** [comment]
Added `max-width` for that feedback message to take less space and look a bit better. To introduce buttons to hide a whole column we can add later (I don't have a good idea of how it should look).

**Review by GTabala — CHANGES_REQUESTED** [review]

File: client/src/components/Reports/GeneralTable.js (hunk 2 above)
Comment (g1st) [reply client/src/components/Reports/GeneralTable.js:79]: Does my [previous commit](https://github.com/HadiyahL/spring-action-cleaning/pull/76/commits/83cdb7c92171738aa1d25215877d80de3d8c775c) solve it?

Comment (g1st) [reply client/src/components/Reports/GeneralTable.js:79]: Oh, I can remove those checks everywhere where it is not used in table footer right, one sec.

Comment (GTabala) [reply client/src/components/Reports/GeneralTable.js:71]: Lines 47, 50, 51, 74 don't need to check if it is `tableFooter`.

**Review by GTabala — CHANGES_REQUESTED** [review]

File: client/src/components/Reports/GeneralTable.js [hunk 3]
```diff
@@ -44,11 +44,9 @@ const GeneralTable = ({ data, tableFooter }) => {
...
-							{customer}
-						</td>
-						<td className={tableFooter && "font-weight-bold"}>{branch}</td>
-						<td className={tableFooter && "font-weight-bold"}>
+						<td>{customer}</td>
+						<td>{branch}</td>
+						<td>
 							{tableFooter
 								? ""
 								: formatDate(visit_on, {
```
Comment (GTabala) [inline client/src/components/Reports/GeneralTable.js:52]: I missed this line. :(

**Review by GTabala — APPROVED** [review]

File: client/src/components/Reports/GeneralTable.js (hunk 3 above)
Comment (g1st) [reply client/src/components/Reports/GeneralTable.js:52]: This one needs to check if it's `tableFooter` otherwise it displays 'Invalid Date' on the table.

**Pull request merged**